src/core/CompUnit/Repository/Installation.pm6

```diff
@@ -430,10 +441,10 @@ sub MAIN(:$name, :$auth, :$ver, *@, *%) {
         }
 
         # Sort from highest to lowest by version and api
-        my $sorted-metas := $matching-metas.sort(*.value<api>).sort(*.value<ver>).reverse;
+        my $sorted-metas := $matching-metas.sort({ .value<name> eq $spec.short-name }).sort(*.value<api>).sort(*.value<ver>).reverse;
 
         # There is nothing left to do with the subset of meta data, so initialize a lazy distribution with it
-        my $distributions := $sorted-metas.map(*.kv).map: -> ($dist-id, $meta) { self!lazy-distribution($dist-id, :$meta) }
+        my $distributions = $sorted-metas.map(*.kv).map: -> ($dist-id, $meta) { self!lazy-distribution($dist-id, :$meta) }
```

Comment:
Was this intentional?

Reply:
I'm not sure what you mean -- `.candidates()` return an `Array of Distribution`, not an `Array of Hash`

Reply:
Did you?

Reply:
Yes. Did you have a concern about this change?

Reply:
Not really, I just saw this stuck conversation, thought about pinging in the only issue that seemed unsolved.